src/routes_admin.py

```diff
@@ -0,0 +1,237 @@
+            staff_id = request.form.get('netid')
+            result = db.remove_staff_from_course(cid, staff_id)
+            if none_modified(result):
+                return util.error(f"'{staff_id}' is not a staff")
+            return util.success(f"Successfully removed '{staff_id}'")
+
+        @blueprint.route("/staff/course/<cid>/promote_staff", methods=["POST"])
+        @auth.require_auth
+        @auth.require_admin_status
+        def promote_course_staff(netid, cid):
+            staff_id = request.form.get('netid')
+            if not verify_staff(staff_id, cid):
+                return util.error(f"'{staff_id}' is not a staff")
+            result = db.add_admin_to_course(cid, staff_id)
+            if none_modified(result):
+                return util.error(f"'{staff_id}' is already an admin")
+            return util.success(f"Successfully made '{staff_id}' admin")
+
+        @blueprint.route("/staff/course/<cid>/demote_admin", methods=["POST"])
+        @auth.require_auth
+        @auth.require_admin_status
+        def demote_course_admin(netid, cid):
+            staff_id = request.form.get('netid')
+            if not verify_staff(staff_id, cid) or not verify_admin(staff_id, cid):
+                return util.error(f"'{staff_id}' is not a admin")
+            db.remove_admin_from_course(cid, staff_id)
+            return util.success(f"Successfully removed '{staff_id}' from admin")
+
+        @blueprint.route("/staff/course/<cid>/add_student", methods=["POST"])
+        @auth.require_auth
+        @auth.require_admin_status
+        def add_course_student(netid, cid):
+            new_student_id = request.form.get('netid')
+            if new_student_id is None:
+                return util.error("Cannot find netid field")
+            result = db.add_student_to_course(cid, new_student_id)
+            if none_modified(result):
+                return util.error(f"'{new_student_id}' is already a student")
+            return util.success(f"Successfully added {new_student_id}")
+
+        @blueprint.route("/staff/course/<cid>/remove_student", methods=["POST"])
+        @auth.require_auth
+        @auth.require_admin_status
+        def remove_course_student(netid, cid):
+            netid = request.form.get('netid')
+            result = db.remove_student_from_course(cid, netid)
+            if none_modified(result):
+                return util.error(f"'{netid}' is not a student")
+            return util.success(f"Successfully removed '{netid}'")
+
+        @blueprint.route("/staff/course/<cid>/<aid>/", methods=["POST"])
+        @auth.require_auth
+        @auth.require_admin_status
+        def staff_add_assignment(netid, cid, aid):
+            missing = util.check_missing_fields(request.form,
+                                                *["max_runs", "quota", "start", "end", "config", "visibility"])
+            if missing:
+                return util.error("Missing fields (%s)." % (", ".join(missing)))
```

Comment:
If we're using f-strings everywhere, let's use it here as well.

Reply:
I didn't write the staff_add_assignment/staff_edit_assignment or staff_get_extensions/staff_add_extension. I copied and pasted them from StaffRoutes to AdminRoutes. I can make these changes if you want.